modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+                            case HOURS:
+                                if (element.getValue().equalsIgnoreCase("all")) {
+                                    hours = -1;
+                                } else {
+                                    hours = timeToLong(element.getValue());
+                                }
+                                retentionPeriods.put(TimePeriod.Duration.HOURS, hours);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private Long getPurgeExecutionInterval() {
+        return purgeExecutionInterval;
+    }
+
+    public boolean isPurgingEnabled() {
+        return purgingEnabled;
+    }
+
+    @Override
+    public void run() {
+        long currentTime = System.currentTimeMillis();
+        long purgeTime;
+        Object[] purgeTimes = new Object[1];
+        List<VariableExpressionExecutor> variableExpressionExecutorList = new ArrayList<>();
+        Attribute attribute;
+        VariableExpressionExecutor variableExpressionExecutor;
+        ComplexEventChunk<StateEvent> eventChunk = new ComplexEventChunk<>(true);
+        BaseIncrementalDataPurgingValueStore baseIncrementalDataPurgingValueStore = new
+                BaseIncrementalDataPurgingValueStore(currentTime, streamEventPool);
+        attribute = new Attribute(INTERNAL_AGG_TIMESTAMP_FIELD, Attribute.Type.LONG);
+        variableExpressionExecutor = new VariableExpressionExecutor(attribute, 0, 1);
+        variableExpressionExecutorList.add(variableExpressionExecutor);
+
+        for (Map.Entry<TimePeriod.Duration, Table> entry : aggregationTables.entrySet()) {
+            if (retentionPeriods.get(entry.getKey()) != -1) {
+                Variable leftVariable = new Variable(INTERNAL_AGG_TIMESTAMP_FIELD);
+                leftVariable.setStreamId(entry.getValue().getTableDefinition().getId());
+                Compare expression = new Compare(leftVariable,
+                        Compare.Operator.LESS_THAN, new Variable(INTERNAL_AGG_TIMESTAMP_FIELD));
+                purgeTime = currentTime - retentionPeriods.get(entry.getKey());
+                purgeTimes[0] = purgeTime;
+                StateEvent secEvent = baseIncrementalDataPurgingValueStore.createStreamEvent(purgeTimes);
+                eventChunk.add(secEvent);
+                Table table = aggregationTables.get(entry.getKey());
+                try {
+                    CompiledCondition compiledCondition = table.compileCondition(expression,
+                            matchingMetaInfoHolder(table, attribute), siddhiAppContext, variableExpressionExecutorList,
+                            tableMap, "query1");
+                    LOG.info("Purging has enable for the table: " + table.getTableDefinition().getId() + " with a" +
+                            " retention of timestamp : " + purgeTime);
+
+                    table.deleteEvents(eventChunk, compiledCondition, 1);
```

Comment:
you can only update the stream event with current time and reuse the same compiledCondition for each iteration